test/check-application

```diff
@@ -51,13 +51,88 @@ class TestApplication(testlib.MachineCase):
+        b.wait_present("#containers-containers-filter button")
+        b.click("#containers-containers-filter button")
+        b.wait_visible("#containers-containers-filter .dropdown-menu")
+        b.click("#containers-containers-filter li[data-data='all'] a")
+        b.wait_present("#containers-containers")
+        b.wait_present('#containers-containers tr:contains("busybox:latest") td:contains("running")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest")')
+
+        # delete a stopped container
+        b.click('#containers-containers tbody tr:contains("alpine:latest") td.listing-ct-toggle')
+        b.wait_visible('#containers-containers tbody tr:contains("alpine:latest") + tr button.btn-delete')
+        b.click('#containers-containers tbody tr:contains("alpine:latest") + tr button.btn-delete')
+        with b.wait_timeout(60):
+            b.wait_present(".modal-dialog div")
+        b.wait_visible(".modal-dialog div .btn-danger")
+        b.click(".modal-dialog div .btn-danger")
+        with b.wait_timeout(60):
+            try:
+                b.wait_visible("#containers-containers")
+                b.wait_not_in_text("#containers-containers", "alpine:latest")
+            except:
+                self.confirm()
+                b.wait_visible("#containers")
+                b.wait_not_in_text("#containers-containers", "alpine:latest")
+
+        # delete a running container
+        self.loadAllContainerlist()
+        b.wait_present('#containers-containers tr:contains("busybox:latest")')
+        b.click('#containers-containers tbody tr:contains("busybox:latest") td.listing-ct-toggle')
+        b.wait_visible('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
+        b.click('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
+        with b.wait_timeout(60):
+            b.wait_present(".modal-dialog div")
+        b.wait_visible(".modal-dialog div .btn-danger")
+        b.click(".modal-dialog div .btn-danger")
+        with b.wait_timeout(60):
+                self.confirm()
+                with b.wait_timeout(60):
+                    b.wait_not_in_text("#containers-containers", "busybox:latest")
+
+        # run a container for Stop test
+        m.execute("podman run -d --name swamped-crate busybox sleep 1000")
+        self.loadAllContainerlist()
+        b.wait_present('#containers-containers tr:contains("busybox:latest")')
+        b.wait_visible('#containers-containers tr:contains("busybox:latest")')
+
+        # stop a running container
+        b.click('#containers-containers tbody tr:contains("busybox:latest") td.listing-ct-toggle')
+        b.wait_visible('#containers-containers tbody tr:contains("busybox:latest") + tr:contains("Stop")')
+        b.click('#containers-containers tbody tr:contains("busybox:latest") + tr:contains("Stop")')
+        self.loadAllContainerlist()
+        # TODO: wait for container to stop
```

Comment:
Please implement this - it's the main functionality you are testing here.

Reply:
I can't wait for the stop because after I run ```TEST_AUDIT_NO_SELINUX=1 test/check-application -s -t -v``` and log into 127.0.0.2:9091, the UI looks like generateed by outdated code, not that I'm working on. The test uses a component class I used before, which the stop button has no onclick event, but I changed it later. The old component class is Dropdown.jsx, but my code currently uses DropdownContainer.jsx not Dropdown.jsx, I didn't use Dropdown.jsx anymore. So I delete Dropdown.jsx and did 

```
git rm <path>
git add .
```
It still uses the old version code. Can you help me figure it out?

Reply:
You most certainly don't want `git add .`, but add the new file you are working on. Did you update all the `import` statements for Dropdown.jsx as well? If so, try `git clean -ffdx` and rebuild from scratch, just in case the build system doesn't properly pick up this change.

Reply:
I added ```testlib.wait(lambda: m.execute("podman ps -a | grep -q Exited"), delay=5)```this line, and tests hangd here. I enterned the shell, 
```
[root@localhost ~]# podman ps -a | grep -q Exited
[root@localhost ~]# podman ps -a | grep Exited
6664a9e8e72d   docker.io/library/busybox:latest   sleep 1000   8 minutes ago   Exited (137) 8 minutes ago
```
It looks like the containre really stoped. but testss till hang there.
the error after Enter is 
```
Traceback (most recent call last):
  File "test/check-application", line 110, in testBasic
    testlib.wait(lambda: m.execute("podman ps -a | grep -qExited"), delay=5)
  File "test/common/testlib.py", line 1312, in wait
    raise Error(msg or "Condition did not become true.")
testlib.Error: Condition did not become true.
```

Reply:
There is a space missing between `-q` and `Exited`. So grep (and thus the whole command) would fail on something like "unknown option".

Reply:
sorry, it's just a typo when I deal with markdown in github.  I used
```
testlib.wait(lambda: m.execute("podman ps -a | grep -q Exited")
```, with space. The problem is not beacuse the space.

Reply:
```
lambda: m.execute("podman ps -a | grep -q Exited"
```
has no reply, I changed to ```lambda: m.execute("podman ps -a | grep -c Exited"```